Scripts/Proxies/TwoHandedProxyBase.cs

```diff
@@ -159,23 +174,51 @@ public virtual IEnumerator Start()
             if (trackedObjectInput == null && m_PlayerInput)
                 trackedObjectInput = m_PlayerInput.GetActions<TrackedObject>();
 
-            var leftProxyHelper = m_LeftHand.GetComponent<ProxyHelper>();
-            var rightProxyHelper = m_RightHand.GetComponent<ProxyHelper>();
-            this.ConnectInterfaces(ObjectUtils.AddComponent<ProxyAnimator>(leftProxyHelper.gameObject), leftProxyHelper.rayOrigin);
-            this.ConnectInterfaces(ObjectUtils.AddComponent<ProxyAnimator>(rightProxyHelper.gameObject), rightProxyHelper.rayOrigin);
+            m_PreviousLeftHandPosition = trackedObjectInput.leftPosition.vector3;
+            m_PreviousRightHandPosition = trackedObjectInput.rightPosition.vector3;
         }
 
-        public virtual void OnDestroy() { }
-
         public virtual void Update()
         {
             if (active)
             {
-                m_LeftHand.localPosition = trackedObjectInput.leftPosition.vector3;
+                var leftLocalPosition = trackedObjectInput.leftPosition.vector3;
+                m_LeftHand.localPosition = leftLocalPosition;
                 m_LeftHand.localRotation = trackedObjectInput.leftRotation.quaternion;
 
-                m_RightHand.localPosition = trackedObjectInput.rightPosition.vector3;
+                var rightLocalPosition = trackedObjectInput.rightPosition.vector3;
+                m_RightHand.localPosition = rightLocalPosition;
                 m_RightHand.localRotation = trackedObjectInput.rightRotation.quaternion;
+
+                if (m_SemitransparentLockRequest == null)
+                {
+                    const float minShakeMovementThreshold = 0.001f;
+                    const int movementDetectionIncreaseScalar = 125;
```

Comment:
These values feel a little magical; maybe they should move to the top of the class, if not a settings object somewhere, particularly because currently the shake seems a little inconsistent / hard to trigger, and we might need to adjust these some more to find the best spot.

Reply:
I think what the shake detection is ultimately missing here is that it is a two-part problem:
You want to detect great average velocity over a period of time while the average position over that period of time has not changed much.  The velocity tracker from the toolkit is very well suited for this kind of thing - @dunity would you mind me taking a crack at this?